### PR TITLE
Remove admin role from application credential

### PR DESCRIPTION
**Pull request by bbezak**

### CODE REVIEW

**Review by JohnGarbutt — APPROVED** [review]
Nice, great job.

File: api/azimuth/provider/openstack/api/core.py [hunk 1]
```diff
@@ -225,6 +225,8 @@ def __init__(self, auth_url, token, interface = 'public', verify = True):
         project = json['token'].get('project', {})
         self.project_id = project.get('id')
         self.project_name = project.get('name')
+        roles_data = json['token'].get('roles', [])
+        self.roles = [{'name': role['name']} for role in roles_data]
```
Comment (mkjpryor) [inline api/azimuth/provider/openstack/api/core.py:229]: Personally, I would prefer that this stores the actual thing returned by the token endpoint, i.e. just:

```python
self.roles = json["token"].get("roles", [])
```

That way, we have all the role information if we want it for something else in the future.

File: api/azimuth/provider/openstack/provider.py [hunk 2]
```diff
@@ -1301,13 +1301,16 @@ def cloud_credential(self, name, description):
...
+        roles = self._connection.roles
+        roles_without_admin = [role for role in roles if role['name'] != 'admin']
         user = self._connection.identity.current_user
         existing = user.application_credentials.find_by_name(name)
         if existing:
             existing._delete()
         app_cred = user.application_credentials.create(
             name = name,
             description = description,
+            roles=roles_without_admin,
```
Comment (mkjpryor) [inline api/azimuth/provider/openstack/provider.py:1313]: And this does the conversion into the specific `{"name": "<rolename>"}` syntax required by this API call, e.g.:

```python
app_cred_roles = [
    {"name": role["name"]}
    for role in self._connection.roles
    if role["name"] != "admin"
]
```

**Review by mkjpryor — CHANGES_REQUESTED** [review]
Nice work. Just a minor thing, in case we want to use the roles for something else in future.

File: api/azimuth/provider/openstack/api/core.py (hunk 1 above)
Comment (bbezak) [reply api/azimuth/provider/openstack/api/core.py]: done

File: api/azimuth/provider/openstack/provider.py (hunk 2 above)
Comment (bbezak) [reply api/azimuth/provider/openstack/provider.py]: done

**Review by mkjpryor — APPROVED** [review]
LGTM

**Pull request merged**